src/index.js

```diff
@@ -0,0 +1,84 @@
+const fs = () => window[window._fs_namespace];
+let didInit = false;
+
+const ensureSnippetLoaded = () => {
+  const snippetLoaded = !!fs();
+  if (!snippetLoaded) {
+    throw Error('FullStory is not loaded, please ensure the FullStory snippet is executed before calling FullStory API functions');
+  }
+  return true;
+};
+
+const hasFullStoryWithFunction = (...testNames) => {
+  const functionsCreated = () => testNames.reduce((acc, current) => acc && fs()[current], true);
+  return ensureSnippetLoaded() && functionsCreated();
+};
+
+const wrapFunction = name => (...params) => {
+  if (hasFullStoryWithFunction(name)) {
+    return fs()[name](...params);
+  }
+  console.warn(`FS.${name} not ready`); // eslint-disable-line no-console
+  return null;
+};
+
+const wrappedFS = ['event', 'log', 'getCurrentSessionURL', 'identify', 'setUserVars', 'consent', 'shutdown', 'restart'].reduce((acc, current) => {
+  acc[current] = wrapFunction(current);
+  return acc;
+}, {});
+
+const { event } = wrappedFS;
```

Comment:
We can destructure all of these in a single statement, like this:
```js
const { event, log, etc... } = wrappedFS;
```

But, now that I look again, why are we destructuring in the first place just to export them as a map again? Can we export `wrappedFS` directly instead?

Reply:
They have to be destructured and exported in order to be imported with the right names.

I was thinking that named exports would provide more flexibility to clients whom might want to import individual FS API functions. But... on second thought it makes sense to use a default export.

```
// in module
export default wrappedFS;

//in client
import FullStory from '@fullstory/browser';

// send custom event
FullStory.wrappedFS.event('whateves', {});
```

I'll change it.

Reply:
Ah, I misread the export. I thought we were default exporting a map instead of exposing named exports.